packages/network/test/integration/proxy-connections.test.ts

```diff
@@ -258,42 +253,112 @@ describe('Proxy connection tests', () => {
             )
     })
 
-    it('if caught, failed publish only connections do not clean out existing connections', async () => {
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.PUBLISH, 'publisher')
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node-2', ProxyDirection.PUBLISH, 'publisher')
-        try {
-            await onewayNode.openProxyConnection(StreamPartIDUtils.parse('stream-5#0'), 'non-existing-node', ProxyDirection.PUBLISH, 'publisher')
-        } catch (err) {
-            // no-op
-        }
-
-        // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getOutboundNodesForStreamPart(defaultStreamPartId)).toContainValues(['contact-node', 'contact-node-2'])
-        // @ts-expect-error private
-        expect(onewayNode.streamPartManager.getNeighborsForStreamPart(defaultStreamPartId)).toBeArrayOfSize(0)
+    it('If connection to any proxy node fails setProxies should reject', async () => {
+        expect(() => onewayNode.setProxies(
+            defaultStreamPartId,
+            ['contact-node', 'contact-node-2', 'non-existing-node'],
+            ProxyDirection.PUBLISH,
+            async () => 'publisher')
+        ).rejects.toThrow()
     })
 
     it('If publish only connection is the only stream connection on contact node it will not unsubscribe', async () => {
         await Promise.all([
             waitForEvent(contactNode, NodeEvent.NODE_UNSUBSCRIBED),
             contactNode2.unsubscribe(defaultStreamPartId)
         ])
-        await onewayNode.openProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.PUBLISH, 'publisher')
+        await onewayNode.setProxies(defaultStreamPartId, ['contact-node'], ProxyDirection.PUBLISH, async () => 'publisher')
         await Promise.all([
             waitForEvent(contactNode, NodeEvent.ONE_WAY_CONNECTION_CLOSED),
-            onewayNode.closeProxyConnection(defaultStreamPartId, 'contact-node', ProxyDirection.PUBLISH),
+            onewayNode.setProxies(defaultStreamPartId, [], ProxyDirection.PUBLISH, async () => 'publisher'),
         ])
         // @ts-expect-error private
         expect(contactNode.streamPartManager.isSetUp(defaultStreamPartId)).toBeTrue()
     })
 
+    it('will not connect above set target limit', async () => {
```

Comment:
Maybe should also have a test for situation where `setProxies` is called with a `connectionCount` that is greater than the `nodeIds` array length? Should fail with a validation error?